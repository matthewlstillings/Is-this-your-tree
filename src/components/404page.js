import React from 'react';
import {BrowserRouter, Route, Switch, Link, NavLink} from 'react-router-dom';

const NotFoundPage = () => (
    <div className="not-found">
        <h1 className="not-found-title">Page not found, please return to the home page.</h1>
        <Link className="not-found-link" to='/'>Go home</Link>
    </div>
);

export default NotFoundPage;