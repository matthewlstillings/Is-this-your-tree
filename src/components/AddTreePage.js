import React from 'react';
import { connect } from 'react-redux';
import TreeForm from './TreeForm'

export class AddTreePage extends React.Component {
   
    onSubmit = () => {
        console.log('Submitted');
        window.location.reload();
    };

    render() {
        return (
            <div className="new-tree">
                <h1 className="new-tree__title">Submit New Tree</h1>
                <p className="new-tree__required">* Required Fields</p>
                <TreeForm
                    onSubmit={this.onSubmit}
                />
            </div>
        )
    }
}

