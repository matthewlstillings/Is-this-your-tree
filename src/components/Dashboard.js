import React from 'react';
import TreeList from './TreeList';
import TreeFilters from './TreeFilters';


const DashboardPage = () => (
    <div className="dashboard">
        <TreeFilters />
        <TreeList />
    </div>
);

export default DashboardPage;
