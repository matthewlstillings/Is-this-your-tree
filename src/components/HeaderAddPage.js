import React from 'react';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faLeaf} from '@fortawesome/free-solid-svg-icons';

export const HeaderAddPage = () => (
    <div className="header__container is-add__page">              
            <div className="header__leaf-container is-add__page">
            <a href="/"><FontAwesomeIcon icon={faLeaf} className="header__leaf is-add__page"/></a>
            </div>
    </div>
)

export default HeaderAddPage;