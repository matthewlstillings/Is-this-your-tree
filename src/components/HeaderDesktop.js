import React from 'react';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faLeaf} from '@fortawesome/free-solid-svg-icons';

export const HeaderDesktop = () => (
    <div className="header__container is-desktop">              
        <div className="header__content is-desktop">
            <h1 className="header__title is-desktop">Is This Your Tree?</h1>
            <h3 className="header__subtitle is-desktop">Find your tree from Central Park</h3>
        </div>
        <div className="header__leaf-container is-desktop">
            <div>
                <a href="/"><FontAwesomeIcon icon={faLeaf} className="header__leaf is-desktop"/></a>
            </div>
        </div>
    </div>
)

export default HeaderDesktop;