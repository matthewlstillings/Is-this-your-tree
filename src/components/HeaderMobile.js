import React from 'react';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faLeaf} from '@fortawesome/free-solid-svg-icons';

export const HeaderMobile = (props) => (
    <div className={"header__container is-mobile " + (props.unlocked === true && " unlocked")}>
        <div className={"header__content is-mobile " + (props.unlocked === true && "unlocked")}>
            <h1 className={"header__title is-mobile " + (props.unlocked === true && "unlocked")}>Is This Your Tree?</h1>
            <h3 className={"header__subtitleÂ is-mobile " + (props.unlocked === true && " unlocked")}>Find your tree from Central Park</h3>
        </div>
        <div className={"header__leaf-container is-mobile " + (props.unlocked === true && " unlocked")}>
            <div>
            <a href="/"><FontAwesomeIcon icon={faLeaf} className={"header__leaf is-mobile " + (props.unlocked === true && " unlocked")} onClick={ 
                    () => {
                        props.unlockApp();
                        document.getElementsByClassName('body')[0].classList.add('unlocked')
                    }        
                }/></a>
                <h2 className={"header__leaf-text is-mobile " + (props.unlocked === true && " unlocked")}>Click to start</h2>
            </div>
        </div>
    </div>
)

export default HeaderMobile;
 




    