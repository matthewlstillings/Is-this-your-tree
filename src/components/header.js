import React from 'react';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faLeaf} from '@fortawesome/free-solid-svg-icons';
//import {BrowserRouter, Route, Switch, Link, NavLink} from 'react-router-dom';
//import {connect} from 'react-redux';
//import {startLogout} from '../actions/auth';

export const Header = () => (
    <header className="header">
        <div className="header__content">
            <h1 className="header__title">Is This Your Tree?</h1>
            <h3 className="header__subtitle">Find your tree from Central Park</h3>
        </div>
        <div className="header__leaf-container">
            <div>
                <FontAwesomeIcon icon={faLeaf} className="header__leaf" onClick={ 
                    () => {
                        const header = document.getElementsByClassName('header')[0];
                        header.classList.add('shrink');
                        const leaf = document.getElementsByClassName('header__leaf')[0];
                        leaf.classList.add('rotated');
                        const leafText = document.getElementsByClassName('header__leaf-text')[0];
                        leafText.classList.add('fadeOut');
                        const body = document.getElementsByClassName('body')[0];
                        body.classList.add('unlock');
                    }        
                }
                />
                <h2 className="header__leaf-text">Click to start</h2>
            </div>
        </div>
    </header>
);


export default Header;