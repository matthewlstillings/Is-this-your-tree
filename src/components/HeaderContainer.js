import React from 'react';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faLeaf} from '@fortawesome/free-solid-svg-icons';
import HeaderAddPage from './HeaderAddPage';
import HeaderMobile from './HeaderMobile';
import HeaderDesktop from './HeaderDesktop';
//import {BrowserRouter, Route, Switch, Link, NavLink} from 'react-router-dom';
//import {connect} from 'react-redux';
//import {startLogout} from '../actions/auth';

export class Header extends React.Component {
    state = {
        header: '',
        windowWidth: window.innerWidth,
        unlocked: false
    }
    componentWillMount = () => {
        setInterval(() => {console.log(this.state);},2000);
    }
    renderHeader = () => {
        if (document.URL === 'https://is-this-your-tree.herokuapp.com/add-tree' 
        || document.URL === 'https://is-this-your-tree.herokuapp.com/admin' 
        || document.URL === 'https://is-this-your-tree.herokuapp.com/admin-login' 
        || document.URL === 'https://is-this-your-tree.herokuapp.com/new-admin') {
            this.setState(()=>({header: 'add-tree'}));
        } else if (window.innerWidth >= 794.8) {
            this.setState(()=>({header: 'desktop'}));
        } else {
            this.setState(()=>({header: 'mobile'}));//change this back to mobile
        }
    }
    unlockApp = () => {
        this.setState(()=>({unlocked: true}));
        const body = document.getElementsByClassName('body')[0];
        body.classList.add('unlocked');
        console.log('clicked');
    }
    handleWidthChange = (windowWidth, event) => {
            this.setState({windowWidth: window.innerWidth})
            this.renderHeader();
    }
    componentDidMount = () => {
        window.addEventListener("resize", this.handleWidthChange);
        this.renderHeader();
        if (window.innerWidth >= 794.8) {
            this.unlockApp();
        }
    }
    render() {
        return (
            <header className="header">
                {
                    this.state.header === 'add-tree' ? (
                       <HeaderAddPage /> 
                    ) : (
                        this.state.header === "desktop" ? (
                            <HeaderDesktop />
                        ) : (
                            <HeaderMobile 
                                unlockApp={this.unlockApp}
                                unlocked={this.state.unlocked}
                            
                            />
                        )
                    )
                }
                
            </header>
        )
    }
    
};


export default Header;