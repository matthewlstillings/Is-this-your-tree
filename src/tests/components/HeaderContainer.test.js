import React from 'react';
import {shallow} from 'enzyme';
import { Header } from '../../components/HeaderContainer';





test('should render header container', () => {
    document.body.innerHTML = `
    <div class="body">
        
    </div>
`;
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
});