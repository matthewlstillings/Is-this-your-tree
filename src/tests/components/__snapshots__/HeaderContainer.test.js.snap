// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header container 1`] = `
<header
  className="header"
>
  <HeaderDesktop />
</header>
`;
